pydm/widgets/timeplot.py

```diff
@@ -1,160 +1,255 @@
 from ..PyQt.QtGui import QLabel, QApplication, QColor
-from ..PyQt.QtCore import pyqtSignal, pyqtSlot, pyqtProperty, QTimer
-from pyqtgraph import PlotWidget, ViewBox, AxisItem, PlotItem
-from pyqtgraph import PlotCurveItem
+from ..PyQt.QtCore import pyqtSignal, pyqtSlot, pyqtProperty, QTimer, QObject
+from pyqtgraph import PlotWidget, ViewBox, AxisItem, PlotItem, GraphicsObject, PlotCurveItem
 import numpy as np
 import time
+import json
+from collections import OrderedDict
 from .baseplot import BasePlot
 from .channel import PyDMChannel
+from .. import utilities
+
+
+class TimePlotCurveItem(PlotCurveItem):
+  def __init__(self, channel_address, **kws):
+    if not 'name' in kws or kws['name'] is None:
+      try:
+        name = channel_address.split("://")[1]
+      except IndexError:
+        name = channel_address
+      kws['name'] = name
+    self._bufferSize = 1200
+    self._update_mode = PyDMTimePlot.SynchronousMode
+    self.data_buffer = np.zeros((2,self._bufferSize), order='f',dtype=float)
+    self.connected = False
+    self.points_accumulated = 0
+    self.latest_value = None
+    self.channel = None
+    self.address = channel_address
+    super(TimePlotCurveItem, self).__init__(**kws)
+  
+  def to_dict(self):
+    return OrderedDict([("channel", self.address), ("name", self.name()), ("color", self.color_string)])
+    
+  @property
+  def address(self):
+    if self.channel is None:
+      return None
+    return self.channel.address
+  
+  @address.setter
+  def address(self, new_address):
+    if new_address is None or len(str(new_address)) < 1:
+      self.channel = None
+      return
+    self.channel = PyDMChannel(address=new_address, connection_slot=self.connectionStateChanged, value_slot=self.receiveNewValue)
+  
+  @property
+  def color_string(self):
```

Comment:
Oh... my bad! Eyes got tricked on this one! I got used to type the pyqtproperty that I see property and automatically relate to pyqtproperty!